benchmarks/eval_b_test.go

```diff
@@ -0,0 +1,325 @@
+)
+
+func BenchmarkOne(b *testing.B) {
+	filePath, err := writePolicyFile(b.Name(), 1, 1, simplePolicyWriter, simpleRolePolicyWriter)
+	if err != nil {
+		b.Fatalf("unable to write to policy file because of error %s", err)
+	}
+	defer os.Remove(filePath)
+
+	rc := ads.RequestContext{
+		Subject: &ads.Subject{
+			Principals: []*ads.Principal{
+				{
+					Type: "user",
+					Name: "user1-1",
+				},
+			},
+		},
+		ServiceName: "bench",
+		Action:      "read",
+		Resource:    "/books/book1",
+	}
+
+	runTest(b, rc, filePath, true)
+}
+
+func BenchmarkTiny(b *testing.B) {
+	filePath, err := writePolicyFile(b.Name(), 10, 10, simplePolicyWriter, simpleRolePolicyWriter)
+	if err != nil {
+		b.Fatalf("unable to write to policy file because of error %s", err)
+	}
+	defer os.Remove(filePath)
+
+	rc := ads.RequestContext{
+		Subject: &ads.Subject{
+			Principals: []*ads.Principal{
+				{
+					Type: "user",
+					Name: "user5-7",
+				},
+			},
+		},
+		ServiceName: "bench",
+		Action:      "read",
+		Resource:    "/books/book5",
+	}
+
+	runTest(b, rc, filePath, true)
+}
+
+func BenchmarkSmall(b *testing.B) {
+	filePath, err := writePolicyFile(b.Name(), 100, 10, simplePolicyWriter, simpleRolePolicyWriter)
+	if err != nil {
+		b.Fatalf("unable to write to policy file because of error %s", err)
+	}
+	defer os.Remove(filePath)
+
+	rc := ads.RequestContext{
+		Subject: &ads.Subject{
+			Principals: []*ads.Principal{
+				{
+					Type: "user",
+					Name: "user50-7",
+				},
+			},
+		},
+		ServiceName: "bench",
+		Action:      "read",
+		Resource:    "/books/book50",
+	}
+
+	runTest(b, rc, filePath, true)
+}
+
+func BenchmarkMedium(b *testing.B) {
+	filePath, err := writePolicyFile(b.Name(), 1000, 10, simplePolicyWriter, simpleRolePolicyWriter)
+	if err != nil {
+		b.Fatalf("unable to write to policy file because of error %s", err)
+	}
+	defer os.Remove(filePath)
+
+	rc := ads.RequestContext{
+		Subject: &ads.Subject{
+			Principals: []*ads.Principal{
+				{
+					Type: "user",
+					Name: "user500-7",
+				},
+			},
+		},
+		ServiceName: "bench",
+		Action:      "read",
+		Resource:    "/books/book500",
+	}
+
+	runTest(b, rc, filePath, true)
+}
+
+func BenchmarkLarge(b *testing.B) {
+	filePath, err := writePolicyFile(b.Name(), 10000, 10, simplePolicyWriter, simpleRolePolicyWriter)
+	if err != nil {
+		b.Fatalf("unable to write to policy file because of error %s", err)
+	}
+	defer os.Remove(filePath)
+
+	rc := ads.RequestContext{
+		Subject: &ads.Subject{
+			Principals: []*ads.Principal{
+				{
+					Type: "user",
+					Name: "user5000-7",
+				},
+			},
+		},
+		ServiceName: "bench",
+		Action:      "read",
+		Resource:    "/books/book5000",
+	}
+
+	runTest(b, rc, filePath, true)
+}
+
+func BenchmarkHuge(b *testing.B) {
+	filePath, err := writePolicyFile(b.Name(), 100000, 10, simplePolicyWriter, simpleRolePolicyWriter)
+	if err != nil {
+		b.Fatalf("unable to write to policy file because of error %s", err)
+	}
+	defer os.Remove(filePath)
```

Comment:
So the benchmark contains the time of removing file?

Reply:
calls b.StopTimer() at the end of each case.